### PR TITLE
ENH Check canViewFile permissions before automatically grant access

### PR DESCRIPTION
**Pull request by sabina-talipova**
### Description
Add condition to check if user has access to view the file and only then give him a session grant access to the file.

### Parent Issue
- https://github.com/silverstripe/silverstripe-assets/issues/501

### CODE REVIEW

File: src/File.php [hunk 1]
```diff
@@ -893,6 +893,9 @@ public function getAbsoluteURL()
      */
     public function getURL($grant = true)
```
Comment (ChloeSartorelli) [inline src/File.php]: The acceptance criteria in the issue say this should not automatically grant by default - same as `getSourceURL()`.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Is this meant to still be in draft?

One of the acceptance criteria for the issue this is attached with is to check for anything else that grants permissions - have you done that?

File: src/File.php (hunk 1 above)
Comment (sabina-talipova) [reply src/File.php]: There is some discussion (https://github.com/silverstripe/silverstripe-assets/issues/501#issuecomment-1207497520) in the parent issue regarding best approach . 
If I understood properly this :
"**_The second option - checking permissions before providing a session grant - seems like a good default as it preserves the intent of the original implementation while clearing out bad cases._**"
I left argument `$grant` as `true` and check does user have a permission to the file or not. I also mentioned this in my comment here https://github.com/silverstripe/silverstripe-assets/issues/501#issuecomment-1248874115

This is the reason why I left this task in Draft for now.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
This is for CMS5 so it needs to target the `2` branch.  You'll need to
a) update the target branch for this PR
b) rebase your work on top of `2`

File: src/File.php (hunk 1 above)
Comment (emteknetnz) [reply src/File.php]: Must be `$grant = false` - the entire purpose of this original issue is to change the default of the optional param to `false`

Comment (sabina-talipova) [reply src/File.php]: DONE

**Comment by sabina-talipova** [comment]
CLOSED after rebase.

**Pull request closed without merging**